drgn/helpers/linux/net.py

```diff
@@ -77,6 +78,24 @@ def get_net_ns_by_fd(task: Object, fd: IntegerLike) -> Object:
     return get_net_ns_by_inode(fget(task, fd).f_inode)
 
 
+_S_IFMT = 0o170000
+_S_IFSOCK = 0o140000
+
+
+def get_socket_by_inode(inode: Object) -> Object:
```

Comment:
It looks like the corresponding function in the kernel is named [`SOCKET_I`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/net/sock.h?h=v5.16-rc6#n1461). drgn helpers usually have the same name as the matching kernel function when possible, what do you think about renaming this to `SOCKET_I`?

There's also [`SOCK_INODE`](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/include/net/sock.h?h=v5.16-rc6#n1466) for going in the opposite direction, should we add that, too?

Reply:
Yes I thought about that, but `SOCKET_I` doesn't check `inode.i_mode`.  See [this](https://elixir.bootlin.com/linux/latest/source/net/netlink/af_netlink.c#L1173):
```c
	if (!S_ISSOCK(inode->i_mode))
		return ERR_PTR(-ENOTSOCK);

	sock = SOCKET_I(inode)->sk;
```
Should we implement both `S_ISSOCK` and `SOCKET_I` instead?  I'm fine with both.

> There's also SOCK_INODE for going in the opposite direction, should we add that, too?

I guess we can just do `sock.file.f_inode`, but let me double check that.

Reply:
I think it's okay if our version of `SOCKET_I` checks `inode.i_mode` even if the kernel version doesn't.

The kernel's `SOCK_INODE` does this:

```c
static inline struct inode *SOCK_INODE(struct socket *socket)
{
	return &container_of(socket, struct socket_alloc, socket)->vfs_inode;
}
```

You probably know more than me about whether that's better or worse than `sock.file.f_inode`.